Add remote control copy for objects (works if no objects are selected…

tsmock opened the pull request:
…, but just zooms to area)

Signed-off-by: Taylor Smock <taylor.smock@kaart.com>


----

It would be nice to have a feature that copies a remote control URL for the objects in the viewport and the viewport itself. This should make it slightly easier to point someone at a specific area with respect to an object.

tsmock committed:
Add remote control copy for objects (works if no objects are selected, but just zooms to area)

Signed-off-by: Taylor Smock <taylor.smock@kaart.com>

tsmock commented:
It also copies multiple urls to clipboard when multiple objects are selected.

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -1,26 +1,34 @@
             Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
             clipboard.setContents(selection, selection);
-            new Notification(tr("Copy: " + linkUser)).setIcon(JOptionPane.INFORMATION_MESSAGE).setDuration(Notification.TIME_SHORT).show();
+            new Notification(tr(COPY, linkUser)).setIcon(JOptionPane.INFORMATION_MESSAGE).setDuration(Notification.TIME_SHORT).show();
```
Translations do not work this way, you have to mark strings for translations using `marktr()`

tsmock replied:
I'll fix this.

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -45,11 +53,11 @@ static void openinBrowserUserOsmComments(String user) {
     public static void copyChangeset(String idChangeset) {
         if (!idChangeset.isEmpty()) {
-            String linkchangeset = "https://www.openstreetmap.org/changeset/" + idChangeset;
+            String linkchangeset = "https://www.openstreetmap.org/changeset/".concat(idChangeset);
```
why this change?

tsmock replied:
I think one of my lint tools was complaining about it (Sonarlint maybe?)
I can change it back, if needed.

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -64,26 +72,75 @@ public static void openinBrowserChangesetMap(String idChangeset) {
+    public static void copyRemoteControl(List<AllOsmObjInfo> osmObjInfo) {
+    	if (osmObjInfo == null || osmObjInfo.isEmpty()) return;
+    	StringBuilder builder = new StringBuilder();
+        builder.append("http://127.0.0.1:8111/load_and_zoom?");
```
instead of hardcoding it, it should be retrieved from JOSM core

tsmock replied:
Fair enough. I'll make the change.

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -64,26 +72,75 @@ public static void openinBrowserChangesetMap(String idChangeset) {
-        if (typeObj != null && !idobj.isEmpty()) {
-            openInBrowser("https://www.openstreetmap.org/" + typeObj + "/" + idobj);
+    public static void openinBrowserIdobj(List<AllOsmObjInfo> osmObjInfo) {
+        if (osmObjInfo.size() > 5 || osmObjInfo.isEmpty()) return;
```
why 5 ?

tsmock replied:
I picked it arbitrarily. I should probably change it to be a config option.

tsmock commented:
Ping

don-vip commented:
@tsmock there are now conflicts can you please resolve them?

tsmock committed:
OSMCha: Update URL (osmcha.mapbox.com -> osmcha.org)

Signed-off-by: Taylor Smock <tsmock@fb.com>

tsmock commented:
The conflicts should be resolved.

Note: I went ahead and updated the osmcha link (`osmcha.mapbox.com` -> `osmcha.org`)

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -39,17 +52,17 @@ public static void openinBrowserUserNeis(String user) {
     public static void copyChangeset(String idChangeset) {
         if (!idChangeset.isEmpty()) {
-            String linkchangeset = "https://www.openstreetmap.org/changeset/" + idChangeset;
+            String linkchangeset = "https://www.openstreetmap.org/changeset/".concat(idChangeset);
```
Config.getUrls().getBaseBrowseUrl() should be used instead of hardcoding https://www.openstreetmap.org

tsmock replied:
Fair enough. I think I was just reusing the pre-existing code.

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -61,29 +74,93 @@ public static void openinBrowserChangeset(String idChangeset) {
+        String linkobjid = "";
+        StringBuilder builder = new StringBuilder();
+        for (AllOsmObjInfo object : osmObjInfo) {
+            builder.append("https://www.openstreetmap.org/")
```
Config.getUrls().getBaseBrowseUrl() should be used instead of hardcoding https://www.openstreetmap.org

don-vip commented on src/org/openstreetmap/josm/plugins/osmobjinfo/OSMObjInfoActions.java:
```diff
@@ -61,29 +74,93 @@ public static void openinBrowserChangeset(String idChangeset) {
+            return;
+        for (AllOsmObjInfo info : osmObjInfo) {
+            if (info.typeObj != null && !info.idObject.isEmpty()) {
+                openInBrowser("https://www.openstreetmap.org/" + info.typeObj + "/" + info.idObject);
```
Config.getUrls().getBaseBrowseUrl() should be used instead of hardcoding https://www.openstreetmap.org

tsmock committed:
OsmObjInfoActions: Use Config.getUrls() instead of hardcoded osm.org

Signed-off-by: Taylor Smock <tsmock@fb.com>

tsmock committed:
OsmObjInfoActions: Update user OSM comments URL

Mapbox appears to have discontinued their osm-comments service.

This replaces that service URL with one from neis-one.org.

Signed-off-by: Taylor Smock <tsmock@fb.com>

tsmock commented:
OK. I've switched everything to the Config.getUrls(), and I also changed the URL for changeset comments. Apparently Mapbox discontinued the osm-comments service, so I just replaced it with a link for neis-one.org.
